libraries/AP_HAL_ChibiOS/hwdef/QioTekZealotF427/hwdef.dat

```diff
@@ -0,0 +1,229 @@
+# hw definition file for processing by chibios_hwdef.py
+# for QioTekZealotF427 hardware from Qio-tek.com 
+
+# MCU class and specific type
+MCU STM32F4xx STM32F427xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1301
+
+FLASH_SIZE_KB 2048
+FLASH_RESERVE_START_KB 128
+
+env OPTIMIZE -O2
+
+# crystal frequency
+OSCILLATOR_HZ 24000000
+
+# this is the STM32 timer that ChibiOS will use for the low level
+# driver. This must be a 32 bit timer. We currently only support
+# timers 2, 3, 4, 5 and 21. See hal_st_lld.c in ChibiOS for details
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 5
+
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+
+# this is the pin that senses USB being connected. It is an input pin
+# setup as OPENDRAIN
+PA9 VBUS INPUT OPENDRAIN
+
+# The normal usage of this ordering is:
+# 1) SERIAL0: console (primary mavlink, usually USB)
+# 2) SERIAL1: telem1
+# 3) SERIAL2: telem2 (recommend ESP8266)
+# 4) SERIAL3: primary GPS
+# 5) SERIAL4: telem3 or GPS2
+# 6) SERIAL5: extra UART or sbus output (usually RTOS debug console)
+
+SERIAL_ORDER OTG1 USART2 USART1 USART3 UART4 UART5
+
+# UART1 
+PB6 USART1_TX USART1
+PB7 USART1_RX USART1
+
+# USART2 for Mavlink wifi module set baudrate to 921600
+PD5 USART2_TX USART2 NODMA
```

Comment:
Disabling DMA on most of the UARTS doesn't seem great to me - is it necessary?

Reply:
It's necessary,STM32F427VIT6 used for the target.  Almost all the resources are used up.